From 6ef7023c0dcfde320015ab19e0e0d423921be77d Mon Sep 17 00:00:00 2001
From: Vincent Chen <vincent.chen@sifive.com>
Date: Mon, 15 Nov 2021 03:39:07 -0800
Subject: [PATCH 1/2] board: sifive: Set LED's color to blue before jumping to
 Linux

The LED's color wil be changed from purple to blue before executing
the sysboot command. Because the sysboot command includes the image loading
from the boot partition, It means the LED's color is blue when executing
"Retrieving file: /Image.gz".

Upstream-Status: Pending
Signed-off-by: Vincent Chen <vincent.chen@sifive.com>
---
 include/configs/sifive-unmatched.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/include/configs/sifive-unmatched.h
+++ b/include/configs/sifive-unmatched.h
@@ -49,6 +49,11 @@
 	"type_guid_gpt_system=" TYPE_GUID_SYSTEM "\0" \
 	"partitions=" PARTS_DEFAULT "\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
-	BOOTENV
+	"setled_blue=mw.l 0x10020024 0x0000ffff; mw.l 0x10020028 0x0000ffff; mw.l 0x1002002c 0x0\0" \
+	BOOTENV \
+	"boot_extlinux=" \
+		"run setled_blue; " \
+		"sysboot ${devtype} ${devnum}:${distro_bootpart} any " \
+			"${scriptaddr} ${prefix}${boot_syslinux_conf};\0"
 
 #endif /* __SIFIVE_UNMATCHED_H */
